packages/server/src/next/app-route-handler.ts

```diff
@@ -27,18 +28,25 @@ export default function factory(
             return NextResponse.json({ message: 'unable to get prisma from request context' }, { status: 500 });
         }
 
+        let params: Context['params'];
         const url = new URL(req.url);
         const query = Object.fromEntries(url.searchParams);
 
-        if (!context.params.path) {
+        try {
+            params = await context.params;
+        } catch {
+            return NextResponse.json({ message: 'Failed to resolve request parameters' }, { status: 500 });
+        }
+
+        if (!params.path) {
             return NextResponse.json(
                 { message: 'missing path parameter' },
                 {
                     status: 400,
                 }
             );
         }
-        const path = context.params.path.join('/');
+        const path = params.path.join('/');
```

Comment:
This is fine, because running `await` on a non-promise also returns the value immediately. It's not required to check if the value is indeed a promise.